### PR TITLE
Quantile Regression Loss Implemented 

### LINKED ISSUE #38035: Quantile Regression Loss
When interested in fitting an unknown probability distribution using a neural network model, we are sometimes interested in quantile regression. I suggest we add the quantile regression loss to the losses we have in pytorch. Example implementation [here](https://medium.com/the-artificial-impostor/quantile-regression-part-2-6fdbc26b2629).

### PR DESCRIPTION
**Pull request by maxmarketit**
Fixes #38035

The simplest way of implementing quantile regression loss I can think of.

Since it is similar to l1loss, I named it q1loss.

We can think of q2loss(which is usually called expectile...)

Seems to work okay

image

image

image

### CODE REVIEW

**Comment by Demetrio92** [comment]
@maxmarketit can you update the PR and clean it up? Is it still relevant? I am looking into quantile regression with PyTorch, and this seems to be the way

**Comment by Demetrio92** [comment]
@apaszke in principle, this PR seems to be fixing #38035, is there a reason it got no attention?
